### ISSUE TITLE
What was the project's inspiration and why it had stagnated?

### ISSUE THREAD

**Issue by amerlyq** [hluk/infinitecopy#1]
This is a very nice and simple little clipboard manager!
Very cool and complete learning example.

Background: recently I looked through source code of basically all (~100) python projects present on github related to clipboard/manager/history (to integrate and automate my regular tasks on Linux) -- and all other codebases are simply trashy: having unsurmountable amateur issues with complexity, hiding bugs in each odd corner case, and littered with assortment of hacks copied from web without understanding.

1) You mentioned this app state is only "alpha" -- but then what was the original idea and roadmap of this application?

2) And if it was abandoned due to some frustrating low-level issues (like X11 message racings) -- could you, please, share your postmortem, so there was no need to re-discover all issues again?)

**Comment by hluk** [comment]
**The main ideas** were to try replace/improve-on [CopyQ](https://github.com/hluk/CopyQ) clipboard manager:

- Use **Python** to allow easier code changes and potentially enable **more contributors** (as opposed to C++, CMake). This could also allow much easier **scripting** (extensions/plugins).
- Use **QML** to have simple and easy-to-change **UI** and possibly list widget with very **large number of items** (this is not supported in Qt without using QML or custom code).
- Use common **database format and SQL** to load items on demand and have more effective search (instead of using custom data format, load it to memory as a whole and use custom code for non-blocking search).

It's been a long time since I wrote any code for this project. I probably stopped contributing because I did not have time and sufficient reasons to continue.

**Problems** I can foresee:
- **Global/system shortcuts** would need to be handled in the desktop environment. I don't know if there is a library I could easily use in Python.
- **Wayland** integration in Qt is still lacking - it could be more difficult to improve this in Python.
- Qt does not support **notifications**.
- There is some custom Objective-C code in CopyQ to fix **clipboard access in macOS**. Again, I'm not sure how much effort would it be to integrate this in Python app.

On the other hand, there is another proof-of-concept Qt+Python app I use as [pomodoro timer](https://github.com/hluk/xitomatl). The really nice feature is that it is automatically compiled into self-sufficient executables for Linux, Windows and macOS using **GitHub Actions**. A minor downside is that currently it includes lots of unneeded Qt libraries and plugins so the size for this simple app is >20MiB.

I'm not sure in what state I left this repo, but I would be interested to look into this further at least to have a simple cross-platform Qt app to monitor clipboard and store into list of items that could be filtered.
